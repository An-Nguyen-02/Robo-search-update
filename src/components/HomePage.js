import React from 'react';
import Cardlist from './Cardlist.js';
import Searchbox from './Searchbox.js';
import './HomePage.css';
import Scroll from './Scroll.js';
import ErrorBoundary from './ErrorBoundary.js';
import Header from './Header.js'


class HomePage extends React.Component {
	componentDidMount(){
		this.props.OnRequestRobots();
	}
	filteredrobots = () =>{
		return this.props.robots.filter(robot =>
								robot.name.toLowerCase().includes(this.props.searchfield.toLowerCase()))
	}
	render(){
		const {OnSearchChange, robots, isPending} = this.props;
		return isPending ?
			<h1>Loading</h1>:
			(
			<div className="tc">
				<Header/>
				<Searchbox searchChange={OnSearchChange}/>
				<Scroll>
					<ErrorBoundary>
	      				<Cardlist robots={this.filteredrobots}/>
	      			</ErrorBoundary>
	      		</Scroll>
	  		</div>
		);

	}
}

export default HomePage