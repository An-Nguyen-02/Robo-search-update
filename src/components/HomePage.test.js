import React from 'react';
import { shallow, mount, render } from 'enzyme';
import HomePage from './HomePage.js';

describe('isPending:false',()=>{
	let wrapper;
	beforeEach(()=>{
		const mockProps = {
			OnRequestRobots: jest.fn(),
			robots: [],
			searchfield: '',
			isPending: false
		}
		wrapper = shallow(<HomePage {...mockProps}/>)
	})

	it('render HomePage component',()=>{
		expect(wrapper).toMatchSnapshot();
	})

	it('filters return none',()=>{
		const mockProps2 = {
			OnRequestRobots: jest.fn(),
			robots: [{
				id: 2,
				name: 'An',
				email: 'An@gmail.conm'
			}],
			searchfield: 'amma',
			isPending: false
		}
		let wrapper2 = shallow(<HomePage {...mockProps2}/>)
		expect(wrapper2.instance().filteredrobots()).toEqual([])
	})

	it('filters return robot',()=>{
		const mockProps3 = {
			OnRequestRobots: jest.fn(),
			robots: [{
				id: 2,
				name: 'An',
				email: 'An@gmail.conm'
			}],
			searchfield: 'an',
			isPending: false
		}
		let wrapper3 = shallow(<HomePage {...mockProps3}/>)
		expect(wrapper3.instance().filteredrobots()).toEqual([{
				id: 2,
				name: 'An',
				email: 'An@gmail.conm'
			}])
	})
})

describe('isPending:true',()=>{
	const mockProps = {
			OnRequestRobots: jest.fn(),
			robots: [],
			searchfield: '',
			isPending: true
		}
		const wrapper = shallow(<HomePage {...mockProps}/>)
		expect(wrapper.find('h1').length).toEqual(1);
})
