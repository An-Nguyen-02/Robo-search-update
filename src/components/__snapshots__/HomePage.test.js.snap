// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isPending:false render HomePage component 1`] = `
<div
  className="tc"
>
  <Header />
  <Searchbox />
  <Scroll>
    <ErrorBoundary>
      <Cardlist
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
